src/core/homepage_elements/featured/hooks.py

```diff
@@ -1,10 +1,30 @@
-from core.homepage_elements.featured import models
+from core.homepage_elements.featured import models, logic
 
 
 def yield_homepage_element_context(request, homepage_elements):
-    if homepage_elements is not None and homepage_elements.filter(name='Featured Articles').exists():
-        featured_articles = models.FeaturedArticle.objects.filter(journal=request.journal)
+    if homepage_elements is not None and homepage_elements.filter(
+            name='Featured Articles',
+    ).exists():
 
-        return {'featured_articles': featured_articles}
+        most_downloaded, number, time = logic.get_popular_article_settings(
+            request.journal
+        )
+
+        if most_downloaded:
```

Comment:
this means we can't have both most downloaded and featured articles at the same time.
Perhaps the "most downloaded" should be a homepage element of its own, so you can choose to have, neither, one, the other or both